Mixmatch/streetview_dataset/parse_TFRecord.ipynb

```diff
@@ -0,0 +1,221 @@
+    "\n",
+    "                if label:\n",
+    "                    img = img.crop(bbox)\n",
+    "                img = img.resize((64, 64))\n",
+    "\n",
+    "                example = img2example(img, label)\n",
+    "                writer.write(example.SerializeToString())\n",
+    "            except Exception as e:\n",
+    "                print(e)\n",
+    "                print(img_path + \" is not valid\")\n",
+    "\n",
+    "def write_TFRecord_TC11(folder_path, writer):\n",
+    "    for img_name in os.listdir(folder_path):\n",
+    "        label = 1\n",
+    "        img_path = os.path.join(folder_path, img_name)\n",
+    "        \n",
+    "        try:\n",
+    "            img = Image.open(img_path, \"r\")\n",
+    "\n",
+    "            # Exclude all non RGB images\n",
+    "            if len(img.getbands()) != 3:\n",
+    "                continue\n",
+    "            \n",
+    "            img = img.resize((64, 64))\n",
+    "\n",
+    "            example = img2example(img, label)\n",
+    "            writer.write(example.SerializeToString())\n",
+    "        except Exception as e:\n",
+    "            print(e)\n",
+    "            print(img_path + \" is not valid\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# input tfrecord file path\n",
+    "input_record_path = './streetlearn-detections/'\n",
+    "UCF_img_dir = './UCF_Streetview_Dataset/raw/'\n",
+    "TC11_img_dir = './TC11/svt1'\n",
+    "\n",
+    "# name format of the output tfrecord files\n",
+    "output_record_path = \"../ML_DATA/\"\n",
+    "train_record_fileName = \"streetview-train.tfrecord\"\n",
+    "test_record_fileName = \"streetview-test.tfrecord\"\n",
+    "\n",
+    "# tfrecord file writer\n",
+    "train_writer = tf.io.TFRecordWriter(output_record_path + train_record_fileName)\n",
+    "test_writer = tf.io.TFRecordWriter(output_record_path + test_record_fileName)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write train and test tfrecord for TC11 dataset\n",
+    "train_image_path = os.path.join(TC11_img_dir, 'train')\n",
+    "test_image_path = os.path.join(TC11_img_dir, 'test')\n",
+    "\n",
+    "write_TFRecord_TC11(train_image_path, train_writer)\n",
+    "write_TFRecord_TC11(test_image_path, test_writer)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write train and test tfrecords for UCF dataset\n",
+    "for i in range(900):\n",
+    "    file_name = \"./streetlearn_detections_tfexample-\" + str(i).zfill(5) + \"-of-01000.tfrecord\"\n",
+    "    parsed_image_dataset = read_TFRecord(os.path.join(input_record_path, file_name))\n",
+    "    write_TFRecord_UCF(parsed_image_dataset, UCF_img_dir, train_writer)\n",
+    "\n",
+    "for i in range(900,1000):\n",
```

Comment:
Add a global constant variable for 1000. Use 0.9*1000 to get the train/test split or add another global constant for 900.

Reply:
Done